framework/sandstone.cpp

```diff
@@ -3630,59 +2831,59 @@ int main(int argc, char **argv)
     if (sApp->total_retest_count < -1 || sApp->retest_count == 0)
         sApp->total_retest_count = 10 * sApp->retest_count; // by default, 100
 
-    if (unsigned(thread_count) < unsigned(sApp->thread_count))
-        restrict_topology({ 0, thread_count });
-    slice_plan_init(max_cores_per_slice);
+    if (unsigned(opts.thread_count) < unsigned(sApp->thread_count))
+        restrict_topology({ 0, opts.thread_count });
+    slice_plan_init(opts.max_cores_per_slice);
```

Comment:
By moving the options into `opts`, now the compiler can't perform dead code elimination based on the values of `thread_count`, `max_cores_per_slice`, etc.

For the cases where we call out-of-line functions (like `debug_init_global` below), there's little code to be eliminated. For everything that gets used here in `main()`, please implement like this:
* rename the variable in `SandstoneOpts` to be `m_varname` or `varname_`
* add a function with the front-end name that is like:
```c++
    int thread_count() const { if (SandstoneOptions::RestrictedCommandLine) return -1; return thread_count_; }
```

Reply:
Question: why not `if constexpr`?